### PR TITLE
feat(responses): drop data error responses if NoSuchChunk is encountered

### PR DESCRIPTION
**Pull request by Yoga07**

### CODE REVIEW

File: src/node/metadata/map_storage.rs [hunk 1]
```diff
@@ -296,6 +297,7 @@ impl MapStorage {
             .map(|data| data.version())
         {
             Ok(res) => Ok(res),
+            Err(Error::NoSuchChunk(addr)) => return Err(Error::NoSuchChunk(addr)),
```
Comment (lionel1704) [inline src/node/metadata/map_storage.rs:300]: What happens to this error? Will the node try to handle it somewhere?

File: src/node/metadata/chunk_records.rs [hunk 2]
```diff
@@ -182,6 +182,12 @@ impl ChunkRecords {
             if !response.is_success() && self.capacity.is_full(&src).await {
                 // We've already responded already with a success
                 // so do nothing
+            } else if !matches!(
+                response,
+                QueryResponse::GetChunk(Err(crate::messaging::client::Error::DataNotFound(_)))
```
Comment (lionel1704) [inline src/node/metadata/chunk_records.rs:187]: If `!matches!(response, Error(...))`
I think the `!` is extra there?

Comment (Yoga07) [reply src/node/metadata/chunk_records.rs]: Ah yes! IDE issues, will be removing 👍

File: src/node/metadata/map_storage.rs (hunk 1 above)
Comment (Yoga07) [reply src/node/metadata/map_storage.rs:300]: We map that error to do nothing [here](https://github.com/Yoga07/sn/blob/5c4668a31e158ea35bd9c36c181b929851c68416/src/node/metadata/mod.rs#L76).

**Review by lionel1704 — APPROVED** [review]

File: src/client/client_api/queries.rs [hunk 3]
```diff
@@ -11,6 +11,8 @@ use crate::client::{connections::QueryResult, errors::Error};
 use crate::messaging::client::{ClientSig, Query};
 use crate::types::{PublicKey, Signature};
 use log::debug;
+use tokio::time::Duration;
+const QUERY_TIMEOUT: u64 = 20; // 20 seconds
```
Comment (connec) [inline src/client/client_api/queries.rs:15]: No opinion on the value,  but `Duration::from_secs` is `const` so you should be able to do:

```rs
const QUERY_TIMEOUT: Duration = Duration::from_secs(20);
```

Comment (Yoga07) [reply src/client/client_api/queries.rs:15]: Oo! Been playing around with that value before pushing, will update :)

File: src/client/client_api/mod.rs [hunk 4]
```diff
@@ -41,6 +42,9 @@ pub struct Client {
     simulated_farming_payout_dot: Dot<PublicKey>,
     incoming_errors: Arc<RwLock<Receiver<CmdError>>>,
     session: Session,
+    query_timeout: Duration,
+    #[cfg(test)]
+    pub override_timeout: Option<Duration>,
```
Comment (lionel1704) [inline src/client/client_api/mod.rs:47]: Doesn't `pub(crate) query_timeout` make this simpler?

**Review by lionel1704 — DISMISSED** [review]

**Review by lionel1704 — APPROVED** [review]

**Review by oetyng — APPROVED** [review]

**Pull request merged**